packages/react-components/lib/tasks/task-timeline.tsx

```diff
@@ -48,47 +47,99 @@ const StyledTimeLine = styled((props: TimeLinePropsWithRef) => <Timeline {...pro
+              <TreeItem
+                key={eventsStates[child].name}
+                nodeId={`child-${id}`}
+                label={eventsStates[id].name}
+              ></TreeItem>
+            );
+          }
+        })}
+      </TreeItem>
+    );
+  }
+}
+
+function colorDot(eventsStates: { [key: string]: EventState } | undefined) {
+  let standby: Boolean = true;
+  for (const id in eventsStates) {
+    const event: EventState = eventsStates[id];
+    if (event.status === 'underway') {
+      return 'success' as const;
+    }
+    if (event.status !== 'standby') {
+      standby = false;
+    }
+  }
+  if (standby) return 'info' as const;
+
+  return 'error' as const;
+}
+
 export interface TaskTimelineProps {
   taskState: TaskState;
 }
 
 export function TaskTimeline({ taskState }: TaskTimelineProps): JSX.Element {
   const phases = taskState.phases ? Object.values(taskState.phases) : [];
-
   return (
     <StyledTimeLine className={classes.timelineRoot}>
-      {phases.map((phase, idx) => (
-        <TimelineItem key={phase.id}>
-          <TimelineOppositeContent style={{ flex: 1, padding: '2px 2px' }}>
-            <Typography variant="overline" color="textSecondary" style={{ textAlign: 'justify' }}>
-              {phase.unix_millis_start_time
-                ? new Date(phase.unix_millis_start_time).toLocaleTimeString()
-                : 'unknown'}
-            </Typography>
-          </TimelineOppositeContent>
-          <TimelineSeparator>
-            <TimelineDot color="info" />
-            {idx < phases.length - 1 && <TimelineConnector />}
-          </TimelineSeparator>
-          <TimelineContent>
-            <TreeView
-              defaultCollapseIcon={<ExpandMoreIcon />}
-              defaultExpandIcon={<ChevronRightIcon />}
-            >
-              {/* FIXME: rmf does not return event hierarchy information */}
-              {phase.events
-                ? Object.values(phase.events).map((event) => (
-                    <TreeItem
-                      key={event.id}
-                      nodeId={event.id.toString()}
-                      label={event.name || 'unknown'}
-                    ></TreeItem>
-                  ))
-                : null}
-            </TreeView>
-          </TimelineContent>
-        </TimelineItem>
-      ))}
+      {phases.map((phase, idx) =>
+        phase.events ? (
+          <TimelineItem key={phase.id}>
+            <TimelineOppositeContent color="text.secondary">
+              <Typography variant="overline" color="textSecondary" style={{ textAlign: 'justify' }}>
+                {phase.original_estimate_millis
+                  ? format(new Date(phase.original_estimate_millis * 1000), "hh:mm aaaaa'm'")
+                  : null}
+              </Typography>
+            </TimelineOppositeContent>
+            <TimelineSeparator>
+              <TimelineDot
+                color={colorDot(phase.events)}
+                sx={{
+                  margin: '0.5px',
```

Comment:
Prefer theme units, also it is weird to specify such a small margin.

Reply:
This small margin was the one I test that resulted with the Dot being connected to the separator to look like the figma figure.
I've removed it for now, I'll add it later while fixing the other small visual issues.